third_party/move/move-vm/runtime/src/session.rs

```diff
@@ -349,6 +349,24 @@ impl<'r, 'l> Session<'r, 'l> {
         Ok(instantiation)
     }
 
+    /// Note: Cannot return a `Function` struct here due to its `pub(crate)` visibility.
+    pub fn load_function_def_is_friend_or_private(
+        &self,
+        module_id: &ModuleId,
+        function_name: &IdentStr,
+        type_arguments: &[TypeTag],
+    ) -> VMResult<bool> {
+        let (_, func, _) = self.move_vm.runtime.loader().load_function(
+            module_id,
+            function_name,
+            type_arguments,
+            &self.data_cache,
+            &self.module_store,
+        )?;
+
+        Ok(func.is_friend_or_private())
+    }
```

Comment:
The name of the function seems a bit too verbose. Can we just call it `is_function_friend_or_private`? Additionally, it should also be possible to implement this check while not requiring the type arguments.

Reply:
On a side note, the reason why Function is `pub(crate)` is because it's an internal data structure and we don't want external (non-vm) clients to depend on it. 

When you need to query information like this, it is recommended that you add specific APIs just to fetch that information, and limit that information to only what's needed. If it turns out in the future that we need to fetch many more kinds of information, we may introduce an API that can return such info as a batch (e.g. `struct FunctionSignature`).

Reply:
I tried avoiding the type arguments, but could not due to function visibility protections in the VM.